### ISSUE TITLE
CollectionShouldNotContainProperty is too aggresive

### ISSUE THREAD

**Issue by jnyrup** [fluentassertions/fluentassertions.analyzers#64]
### Description

The `CollectionShouldNotContainProperty` is triggered when the assertions already seems idiomatic.

An example is when the expression inside `OnlyContain` contains no negations.
```csharp
var list = new[] { "" };

list.Should().OnlyContain(e => e.Contains(""));
```

### Expected behavior:

The analyzer should not be triggered, as the assertion is already idiomatic.

### Actual behavior:

The analyzer is triggered.

### Versions

Fluent Assertions Analyzers 0.11.2
.NET Framework 4.6.1

**Comment by Meir017** [comment]
@jnyrup since the case of a `Should` followed by `OnlyContain` does'n currently have a codefix maybe I should just ignore this case for now untill I fix #2 

In this case the CodeFix should be
```cs
list.Should().NotContain(e => !e.Contains(""));
```
which seems fair

**Comment by jnyrup** [comment]
Maybe I was too quick on this one.
I *personally* prefer `OnlyContain(e => e.MyBooleanProperty)` over `NotContain(e => !e.MyBooleanProperty)` and that's the example in https://fluentassertions.com/tips#Collections-22.

But if someone prefers it the other way around, it is of course beneficial to have a tools to do the transforms.
For many of its refactorings, Roslynator has methods for transforming in both directions, but has one disabled by default.
E.g. 
https://github.com/JosefPihrt/Roslynator/blob/master/docs/refactorings/RR0168.md
and 
https://github.com/JosefPihrt/Roslynator/blob/master/docs/refactorings/RR0171.md
